pkg/controllers/vdb/onlineupgrade_reconciler.go

```diff
@@ -512,6 +509,42 @@ func (r *OnlineUpgradeReconciler) postStartReplicationMsg(ctx context.Context) (
 	return r.postNextStatusMsg(ctx, startReplicationMsgInx)
 }
 
+// prepareReplication makes sure there is at least an Up node in the main cluster
+// and the sandbox, to perform replication.
+// Once we start using services for replication, we will check only the scs served by the services.
+func (r *OnlineUpgradeReconciler) prepareReplication(ctx context.Context) (ctrl.Result, error) {
+	// Skip if the VerticaReplicator has already been created.
+	if vmeta.GetOnlineUpgradeReplicator(r.VDB.Annotations) != "" {
+		return ctrl.Result{}, nil
+	}
+	// we need fresh podfacts just in case some pods went down while
+	// sandbox upgrade was in progress.
+	r.PFacts[vapi.MainCluster].Invalidate()
```

Comment:
Should we invalidate sandbox pfacts too? Don't know how long the draining_connection will take. If the sandbox died after the upgrade but before replication, replication could fail, then we will not do replication any more.

Reply:
I already did it. check out `getSandboxPodFacts()`